lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -1906,6 +1906,14 @@ bool OptimizeReduceMean::run(Function *F, const CompilationContext &cctx) {
           RM->getName().str() + ".transposeNCHW2NHWC", in, NCHW2NHWC, "NHWC");
       auto *AP = F->createAvgPool(RM->getName().str() + ".avgPool", TR1,
                                   kernels, strides, pads);
+      if (AP->getResult().getType()->isQuantizedType()) {
+        auto TypeAP = F->getParent()->uniqueType(
+            AP->getResult().getType()->getElementType(),
+            AP->getResult().getType()->dims(),
+            RM->getResult().getType()->getScale(),
+            RM->getResult().getType()->getOffset());
+        AP->getResult().setType(TypeAP);
+      }
```

Comment:
This is a little messier than it needs to be I think  -- can we just do 
```suggestion
      auto *AP = F->createAvgPool(RM->getName().str() + ".avgPool", TR1,
                                  RM->getResult().getType(), kernels, strides, pads);
```

Reply:
@jfix71 the suggested change of  RM->getResult().getType() is not enough in this case. The dimensions of RM->getResult() and Avg pool output are different. 
Here it will be better if we could develop a new Graph API _uniqueTypeWithNewQuantParams_ which works like this:

auto typeAP = F->getParent()->uniqueTypeWithNewQuantParams(
                                                           AP->getResult().getType(), RM->getResult().getType());

Reply:
Ah right. Adding `uniqueTypeWithNewQuantParams` sounds reasonable to me. Did you want to add it and use it in this PR?

Reply:
Yes, I wanna add this API in this patch itself. I will make the modification soon.